src/transforms/filter.jl

```diff
@@ -0,0 +1,51 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f, table)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter <: Transform
+    f::Base.Callable
+
+
+    function Filter(new_filter)
+        new(new_filter)
+    end
+
+    function Filter()
+        new(_default_filter)
+    end
+end
+
+isrevertible(::Type{Filter}) = true
+
+#Ideally, this function would also work without a given function, as follows:
+#[item for item in iterable if item]
+function apply(transform::Filter, table)
+    indices = [i for i in range(1,length(table)) if transform.f(table[i])]
+    table[indices], (indices, table[_minus(indices, table)])
```

Comment:
Similarly, you cannot index a Tables.jl table. Can you please review their API before continuing the work in the PR?

Reply:
I had written the tests with `TypedTables` and all of them passed on my machine (Ubuntu 20.04). Now I realize these two packages are not necessarily compatible.  
  
Adding `table = Tables.rowtable(input_table)` line allows us to use both the `length` function and indexing, although this always returns a `Vector{NamedTuple}` regardless of the input type. I am not sure if this is the desired outcome.

Reply:
I found the reason why test run fails. I am using Julia 1.7 while the PR runs are on v1.6. I download v.16.5 and tried this, and the difference in behavior does exist. What do I do in this case? Write a code that runs with v1.6 or stick with v1.7?